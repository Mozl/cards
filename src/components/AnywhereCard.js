import React, { Component } from 'react';
import styled from 'styled-components';

const Box = styled.div`
  border: 0.1rem solid black;
  width: 15rem;
  display: flex;
  flex-direction: column;
  text-color: black;
  border-radius: 1.25rem;
  padding-bottom: 1rem;
  margin-bottom: 1rem;
`;

const Wrapper = styled.div`
  margin: 1rem 0 0 1rem;
`;

const CardName = styled.div`
  display: flex;
  padding-left: 0.5rem;
  border: 0.1rem solid black;
  border-radius: 1.25rem;
  margin: 1rem 0 1rem 0;
`;

class AnywhereCard extends Component {
  render() {
    return (
      <Wrapper>
        <CardName onClick={this.props.selectAnywhereCard}>
          <h2>Anywhere Card</h2>
        </CardName>

        {this.props.selected && (
          <Box className="box">
            <p>Apr: 33.9%</p>
            <p>Balance Transfer Offer Duration: 0 months</p>
            <p>Purchase Offer Duration: 0 Months</p>
            <p>Credit Available: Â£300</p>
          </Box>
        )}
      </Wrapper>
    );
  }
}

export default AnywhereCard;
