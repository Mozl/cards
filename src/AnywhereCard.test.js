import React from 'react';
import ReactDOM from 'react-dom';
import AnywhereCard, { CardName } from './components/AnywhereCard';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AnywhereCard />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<AnywhereCard />);
  expect(tree).toMatchSnapshot();
});

it('prop selected changes', () => {
  const wrapper = mount(<AnywhereCard selected={false} />);
  wrapper.setProps({ selected: true });
  expect(wrapper.props().selected).toEqual(true);
});

it('has a credit of 300', () => {
  const wrapper = mount(<AnywhereCard selected={false} anywhereCredit={300} />);
  expect(wrapper.props().anywhereCredit).toBe(300);
});

it('calls onClick', () => {
  const mockOnClick = jest.fn();
  const wrapper = mount(<AnywhereCard onClick={mockOnClick} />);
  wrapper.find(CardName).simulate('click');
  expect(mockOnClick.mock.calls.length).toEqual(1);
});
