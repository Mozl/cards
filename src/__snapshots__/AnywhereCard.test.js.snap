// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bwzfXH uTxCW"
>
  <div
    className="cardName sc-htpNat jfzVHX"
  >
    <h2>
      Anywhere Card
    </h2>
  </div>
  <div
    className="box sc-bdVaJa dIWyBO"
  >
    <p>
      Apr: 33.9%
    </p>
    <p>
      Balance Transfer Offer Duration: 0 months
    </p>
    <p>
      Purchase Offer Duration: 0 Months
    </p>
    <p>
      Credit Available: Â£
    </p>
  </div>
</div>
`;
